src/screens/DiscoveryScreenV2/elements/Search.js

```diff
@@ -109,7 +104,11 @@ const DiscoverySearch = ({
     setLastSearch(text);
     setAllLoading(true);
     setIsFirstTimeOpen(false);
-    fetchDiscoveryData(text);
+    if (fetchData !== undefined) {
+      fetchData(true, text);
+    } else {
+      fetchDiscoveryData(text);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `fetchDiscoveryData` is now conditionally called based on whether `fetchData` is defined. This could potentially change the behavior of the component. Ensure that this conditional logic doesn't break any existing functionality.
```diff
- fetchDiscoveryData(text);
+ if (fetchData !== undefined) {
+   fetchData(true, text);
+ } else {
+   fetchDiscoveryData(text);
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->